src/Parsing.cc

```diff
@@ -109,15 +94,11 @@ xatos(const char *token)
 {
     long port = xatol(token);
 
-    if (port < 0) {
-        debugs(0, DBG_PARSE_NOTE(DBG_IMPORTANT), "ERROR: The value '" << token << "' cannot be less than 0.");
-        self_destruct();
-    }
+    if (port < 0)
+        throw Cfg::FatalError(ToSBuf("invalid value: '", token, "' cannot be less than 0"));
 
-    if (port & ~0xFFFF) {
-        debugs(0, DBG_PARSE_NOTE(DBG_IMPORTANT), "ERROR: The value '" << token << "' is larger than the type 'short'.");
-        self_destruct();
-    }
+    if ((port & ~0xFFFF) != 0)
+        throw Cfg::FatalError(ToSBuf("invalid value: '", token, "' is larger than the type 'short'"));
```

Comment:
```suggestion
        throw Cfg::FatalError(ToSBuf("invalid value: '", token, "' must be a 16-bit unsigned integer"));
```

Reply:
I would rather leave these messages saying what they did prior to this PR for now. All of these functions need a refactoring into libcfg and changes to how they are working can be implemented then.

Reply:
ok